test/acceptance/test/utils.go

```diff
@@ -845,3 +846,18 @@ func getGitProvider(org string, repo string, providerName gitproviders.GitProvid
 
 	return gitProvider, orgRef, err
 }
+
+func setGitProvider() (gitproviders.GitProviderName, string, string) {
```

Comment:
Let's use this logic only once here https://github.com/weaveworks/weave-gitops/blob/main/test/acceptance/test/testsuite_common_test.go#L52 instead of calling it multiple time in each `BeforeEach`.  If you want to keep this function and call it once in `BeforeSuite`  it's fine I would just suggest changing the function name to something more align to its functionality, maybe something like `getGitProviderInfo`,`getGitProviderData` or something similar. Usually methods with `set` in their name are meant to receive parameters and do not return values.